interview-questions.md

```diff
@@ -8,48 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is a melding of JavaScript syntax and HTML syntax, that lets us use both in the same file.  When using one or the other there are slight syntax convention differences in certain situations as a way to reduce conflict within the language.  One of these differences is `class`, which is HTML syntax for assigning a class, and `className` which is the JavaScript syntax for doing the same thing.  `class` is a reserved word in JavaScript, so another had to be chosen.
 
-Researched answer:
+Researched answer: JSX converts HTML tags into React elements. Though I already knew this, I suppose it would have been helpful to add: using `{}` allows us to write javascript code inside HTML elements.  All code that goes into JSX is intended to be transformed into JavaScript.
 
-1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
+2. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package management tool for React. it can be used to install and run things like the "Jest" debugging software.
 
-Researched answer:
+Researched answer: Yarn is allows you to share code with others.
 
 3. What is a React component?
 
-Your answer:
+Your answer: A React component is a separate `.js` file that holds code for an application.  This file is called by the main `App.js` file when its code is needed.  Creating components in React allows us to reduce the amount of code that needs to be run when making a change on a web page.  Clicking a 'like' button would normally cause a full-page refresh to show a change, but putting that action into a single component allows us to only refresh that particular element, and keep the rest of the page static.  Changes are shown when comparing the VirtualDOM to the regular DOM and finding the 'diff', or difference between the two.  Then the DOM is updated to match the VDOM.
 
-Researched answer:
+Researched answer: Components are reusable pieces of code, akin to functions in JavaScript. They can be passed as props to other components.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: props are essentially more advanced parameters that are used in component files.  They're used as vehicles to pass specific values from `App.js` to the components, and get a return.  These values are only precessed when needed, and are not held in a persistent capacity, and the value passing only happens in the downstream direction (away from `App.js`)
```

Comment:
It's important to know that it is not only from App.js to other components. Props can be passed from any component, as long as there is a nested component. Like you said, the value passing happens downstream for conventional props, but props being passed from any component to a nested component within that component is still downstream!

Reply:
Thanks for that clarification. Updated my mental model with that info!